libglusterfs/src/statedump.c

```diff
@@ -282,14 +284,14 @@ gf_proc_dump_xlator_mem_info_only_in_use(xlator_t *xl)
     gf_proc_dump_write("num_types", "%d", xl->mem_acct->num_types);
 
     for (i = 0; i < xl->mem_acct->num_types; i++) {
+#ifdef DEBUG
```

Comment:
Instead of doing this only for debug builds (which would cause production state dumps to contain unnecessary data for types that are not in use), we should check whether `xl->mem_acct->rec[i].num_allocs` is 0 or not.

Reply:
Nice catch!

Reply:
done